documentation/confluent-avro-documentation.md

```diff
@@ -0,0 +1,123 @@
+# ABRiS - Confluent Avro documentation
+
+- [Avro to Spark](#Avro-to-Spark)
+- [Spark to Avro](#spark-to-avro)
+
+## Avro to Spark
+
+### Providing Avro schema    
+```scala
+import za.co.absa.abris.avro.functions.from_confluent_avro
+
+def readAvro(dataFrame: DataFrame, schemaString: String): DataFrame = {
+
+  dataFrame.select(from_confluent_avro(col("value"), schemaString) as 'data).select("data.*")
+}
+```
+The main difference between ```from_confluent_avro``` and ```from_avro``` is in whether it expects the schema_id in the Avro payload. The usage is identical to previous examples.
+
+### Using schema registry
+Schema Registry configuration is the same as in previous Schema Registry example.
+```scala
+import za.co.absa.abris.avro.functions.from_confluent_avro
+
+def readAvro(dataFrame: DataFrame, schemaRegistryConfig: Map[String, String]): DataFrame = {
+
+  dataFrame.select(from_confluent_avro(col("value"), schemaRegistryConfig) as 'data).select("data.*")
+}
+```
+The only difference is the expression name.
+
+       
+### Using Schema Registry for key and value
+In case we are sending Avro data using Kafka we may want to serialize both the key and the value of the Kafka message.
+The serialization of Avro data is not really different when we are doing it for key or for value, but Schema Registry handles each of them slightly differently.
+
+The way the library knows whether you are working with key or value is the schema naming strategy.
+Use ```SchemaManager.PARAM_KEY_SCHEMA_NAMING_STRATEGY``` for key and ```SchemaManager.PARAM_VALUE_SCHEMA_NAMING_STRATEGY``` for value. If the configuration contains **both** of them, it will **throw**!
```

Comment:
Wouldn't this be the solution to `key` boolean? Agree it is not too explicit, though.

Reply:
Yes, but the boolean would allow you to put all the configuration to one map, and then use it. Now you have to create separated maps for key and value.

Reply:
But as I said yesterday, I would prefer to solve the key as a different PR.